### DISCUSSION TITLE
Support for mTLS when connecting to the registry

### DISCUSSION THREAD

**Discussion by manveer94** [aquasecurity/trivy#4631] [Ideas]
### Description

The requirement is to connect to the registry which is behind a reverse proxy server which only accepts mTLS connections.
For making the connection to the registry client certificate has to be provided as per mTLS authentication.

### Target

Container Image

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
How can we implement it?

**Reply by manveer94** [reply]
mTLS requires client certificate when initiating a api call to the registry.
I suggest that we modify the Get function in remote/remote.go to use client certificates when creating the API call.
Certificate can be provided in the Registry Options. 

eg: 
```
if len(option.ClientCert) != 0 && len(option.ClientKey) != 0 {
		var err error
		transport, err = httpTransportWithMtls(option.Insecure, option.ClientCert, option.ClientKey)
		if err != nil {
			return nil, err
		}
	} else {
		transport = httpTransport(option.Insecure)
	}
```

**Reply by knqyf263** [reply]
I think `httpTransport` should handle those client certificates, but it looks good anyway. Created https://github.com/aquasecurity/trivy/issues/4640

**Reply by y4ney** [reply]
Hi! @knqyf263 ,How can I use this feat?There is no flag to pass the value of ClientCert and ClientKey

**Reply by knqyf263** [reply]
It's not exported for CLI users.
